core/audits/seo/link-text.js

```diff
@@ -126,7 +151,21 @@ class LinkText extends Audit {
           return false;
         }
 
-        return BLOCKLIST.has(link.text.trim().toLowerCase());
+        const searchTerm = link.text.trim().toLowerCase();
+
+        if (searchTerm) {
+          // walk through lang code, e. g. "en-US" or "zh-cmn-Hans-CN"
+          let lang = link.textLang;
+          while (lang) {
+            if (nonDescriptiveLinkTexts[lang] &&
+              nonDescriptiveLinkTexts[lang].has(searchTerm)) {
+              return true;
+            }
+            lang = lang.substring(0, lang.lastIndexOf('-'));
```

Comment:
I'm not that deep into linguistics. But valid language codes like `zh-cmn-Hans-CN` exist and there might be differences between e. g. Cantonese and Simplified Chinese. So instead of speculating and to keep it future proof (support more than one segment if needed), I decided to use the loop.